### DISCUSSION TITLE
STM prescaler (clock SPI)

### DISCUSSION THREAD

**Discussion by Laiaborrell** [TheAlbertDev/MASB#55] [Q&A] [answered]
Hi!
I was starting the `.ioc` file configuration and in SPI there is a `prescaler (Baud Rate)` parameter for the clock signal of the SPI. I could not find information of which value should be given to this parameter. Any help?
Pd: it takes values of 2^(x). 
Thanks,
Laia

**Comment by Albert-Alvarez** [comment, marked as answer]
The prescaler serves to divide the clock frequency sourced to the SPI peripheral in order to achieve the desired frequency transmission during the communication.

First, we start finding which is the frequency of the clock signal sourced to the SPI peripheral. From the [datasheet of the STM32F401RE (page 14)](https://www.st.com/resource/en/datasheet/stm32f401re.pdf) we can see that SPI peripherals use the clock APB1 or APB2 (depending on the SPI peripheral used). For example, SPI2 uses APB1.

![image](https://user-images.githubusercontent.com/34403794/116547687-7f3aa080-a8f3-11eb-85f9-031deb56a5e6.png)

Then, we can go to the STM32CubeIDE and check which frequency is configured for the APB2. We can see that is configured with 42 MHz.

![image](https://user-images.githubusercontent.com/34403794/116547979-df314700-a8f3-11eb-91be-1bb21db168a5.png)

If we want a frequency transmission equal to the example in the other issue (500 kHz), we obtain a prescaler equal to 84. We have 84 available as prescaler? No. We can use 64 (frequency transmission equal to 656.25 kHz) or 128 (frequency transmission equal to 328,125). You can choose either of the two as the obtained frequencies are in the [frequency range compatible with the sensor](https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bme280-ds002.pdf).

![image](https://user-images.githubusercontent.com/34403794/116548445-8615e300-a8f4-11eb-8d49-a10148b96bef.png)

**Comment by Laiaborrell** [comment]
Ok! Thank you
